src/Behat/Behat/Context/Snippet/Generator/ContextSnippetGenerator.php

```diff
@@ -174,52 +148,46 @@ private function getMethodArguments(StepNode $step, $tokenCount)
     /**
      * Returns an array of classes used by the snippet template
      *
-     * @param StepNode $step
-     *
      * @return string[]
      */
-    private function getUsedClasses(StepNode $step)
+    private function getUsedClasses(StepNode $step): array
     {
-        $usedClasses = array('Behat\Behat\Tester\Exception\PendingException');
+        $usedClasses = [PendingException::class];
 
         foreach ($step->getArguments() as $argument) {
-            if ($argument instanceof TableNode) {
-                $usedClasses[] = 'Behat\Gherkin\Node\TableNode';
-            } elseif ($argument instanceof PyStringNode) {
-                $usedClasses[] = 'Behat\Gherkin\Node\PyStringNode';
-            }
+            $usedClasses[] = match (true) {
+                $argument instanceof TableNode => TableNode::class,
+                $argument instanceof PyStringNode => PyStringNode::class,
+            };
         }
 
         return $usedClasses;
     }
 
     /**
      * Generates snippet template using regex, method name and arguments.
-     *
-     * @param string   $pattern
-     * @param string   $methodName
-     * @param string[] $methodArguments
```

Comment:
Please keep the info about the $methodArguments parameter as it provides info about the array shape